src/Evolution/Initialization/FunctionOfTimeFromFile.hpp

```diff
@@ -0,0 +1,145 @@
+    // of time
+    const auto& dataset_names{
+        db::get<Initialization::Tags::FuncOfTimeSetNames>(box)};
+
+    // Currently, only support order 3 piecewise polynomials
+    // This could be generalized later, if needed
+    constexpr size_t max_deriv{3};
+
+    std::unordered_map<std::string,
+                       domain::FunctionsOfTime::PiecewisePolynomial<max_deriv>>
+        functions_of_time;
+
+    h5::H5File<h5::AccessType::ReadOnly> file(file_name);
+    for (auto dataset_name : dataset_names) {
+      const auto& dataset = dataset_name[0];
+      const auto& name = dataset_name[1];
+      const auto& dat_file = file.get<h5::Dat>("/" + dataset);
+      const auto& dat_data = dat_file.get_data();
+
+      // Check that the data in the file uses deriv order 3
+      // Column 3 of the file contains the derivative order
+      const size_t dat_max_deriv = dat_data(0, 3);
+      ASSERT(dat_max_deriv == max_deriv,
+             "Deriv order in " << file_name << " should be " << max_deriv
+                               << ", not " << dat_max_deriv);
+
+      // Get the initial time ('time of last update') from the file
+      // and the values of the function and its derivatives at that time
+      const double start_time = dat_data(0, 1);
+
+      // Currently, assume the same number of components are used
+      // at each time. This could be generalized if needed
+      const size_t number_of_components = dat_data(0, 2);
+
+      std::array<DataVector, max_deriv + 1> initial_coefficients;
+      for (size_t i = 0; i < max_deriv + 1; ++i) {
+        gsl::at(initial_coefficients, i) = DataVector(number_of_components);
+        for (size_t a = 0; a < number_of_components; ++a) {
+          // Columns in the file have the following form::
+          // 0 = time
+          // 1 = time of last update
+          // 2 = number of components
+          // 3 = maximum derivative order
+          // 4 = version
+          // 5, ... 5 + maximum derivative order: first component and derivs
+          // ... (sets of coefficients for the next component and derivs)
+          // ...
+          gsl::at(initial_coefficients, i)[a] =
+              dat_data(0, 5 + (max_deriv + 1) * a + i);
+        }
+      }
+      functions_of_time[name] = domain::FunctionsOfTime::PiecewisePolynomial<3>(
+          start_time, initial_coefficients);
+
+      // Loop over the remaining times, updating the function of time
+      DataVector highest_derivative(number_of_components);
+      for (size_t row = 1; row < dat_data.rows(); ++row) {
+        for (size_t a = 0; a < number_of_components; ++a) {
+          highest_derivative[a] =
+              dat_data(row, 4 + (max_deriv + 1) * a + max_deriv);
+        }
+        functions_of_time[name].update(dat_data(row, 1), highest_derivative);
+      }
+    }
+
+    // Create an unordered map of the same type as the FunctionsOfTime tag,
+    // and then add this to the databox
```

Comment:
`databox` -> `DataBox`

Reply:
Comment no longer present because now it's a mutating action, not an adding action